backend/dataall/modules/datasets/services/dataset_table_service.py

```diff
@@ -164,11 +164,11 @@ def sync_existing_tables(session, dataset_uri, glue_tables=None):
 
     @staticmethod
     def _attach_dataset_table_permission(session, dataset: Dataset, table_uri):
-        # ADD DATASET TABLE PERMISSIONS
-        env = EnvironmentService.get_environment_by_uri(session, dataset.environmentUri)
+        """
+        Attach Table permissions to dataset groups
+        """
         permission_group = {
             dataset.SamlAdminGroupName,
-            env.SamlGroupName,
```

Comment:
I see we are removing env admins from getting table permissions for any new dataset tables - but do we clean up the existing permissions for previous tables created?

UPDATE: I actually see we are not cleaning up permissions for tables (maybe a separate PR should address this?). Looking at the way we handle "deleting" tables:

- If `sync()` determines table no longer exists in glue - we just update the table status to `Deleted` and do not show on UI (table item remains along with associated permissions 
  - OK, but could be made better
  - If table is shared a bunch and then sync determines it no longer exists and it disappears it is not straightforward
  - Maybe should keep on UI with a `Deprecated` or `Deleted` Flag telling user to delete table and all associated shares?

- If user deletes table from UI - we are deleting table if no shares exists on that table but not deleting the associated permissions
  - Not as OK - should be cleaning up orphaned permissions

Reply:
New Issue Created - https://github.com/data-dot-all/dataall/issues/1171